cache/lru/cache.go

```diff
@@ -55,15 +72,19 @@ func (c *Cache[K, V]) Get(key K) (V, bool) {
 func (c *Cache[K, _]) Evict(key K) {
 	c.lock.Lock()
 	defer c.lock.Unlock()
-
+	value, ok := c.elements.Get(key)
+	if !ok {
+		return
+	}
+	c.onEvict(key, value)
 	c.elements.Delete(key)
 }
 
 func (c *Cache[_, _]) Flush() {
-	c.lock.Lock()
-	defer c.lock.Unlock()
```

Comment:
I think we still want to hold the lock here to guarantee that `Flush` returns with an empty cache